File: consumer-tests/java-consumer/src/main/java/Consumer.java
import cloud.lingya.agents.sdk.AgentsClient;
import cloud.lingya.agents.sdk.OpenApiCredentials;
import cloud.lingya.agents.sdk.generated.model.AiChatInput;

public final class Consumer {
    private Consumer() {
    }

    public static void compileJavaConsumer() {
        var client = new AgentsClient(
            "https://tenant.example.com",
            "channel",
            new OpenApiCredentials("abcdefghijklmnopqrstuvwxyzABCDEF", "secret")
        );
        var user = client.blockingForUser("external-user");
        var input = new AiChatInput("hello", null, null, null);
        user.getChat().createChat(input);
        input.getQuery();
    }
}

File: consumer-tests/kotlin-consumer/src/main/kotlin/Consumer.kt
import cloud.lingya.agents.sdk.AgentsClient
import cloud.lingya.agents.sdk.OpenApiCredentials
import cloud.lingya.agents.sdk.generated.model.AiChatInput
import cloud.lingya.agents.sdk.generated.model.AiChatStreamInput

fun compileKotlinConsumer(client: AgentsClient) {
    val user = client.forUser("external-user")
    val input = AiChatInput("hello")
    user.chat.streamChatEvents("conversation", AiChatStreamInput("message"))
    println(input.query)
}

fun createKotlinClient(): AgentsClient = AgentsClient(
    "https://tenant.example.com",
    "channel",
    OpenApiCredentials("abcdefghijklmnopqrstuvwxyzABCDEF", "secret"),
)
